### PR TITLE
Maintenance preparations

### PR DESCRIPTION
**Pull request by rpdelaney**
Changes proposed in this pull request:

- Add `requirements.txt`
- Move various invocations of `open()` into context managers
- Fix bare `except:` statements that would have caught system exit, SIGHUP, other variants of `BaseException` unintentionally
- Remove some assignments of variables that were not being used
- Blacken code and begin PEP8 conformance

### CODE REVIEW

**Comment by rpdelaney** [comment]
On line 191 there's still a misuse of `open()` but I don't yet understand what this code is doing there... this PR just has the easy fixes to get started.

**Comment by suisseWalter** [comment]
All the code changes look good. 

Using black as a formater is good. 

Pep8 compliance is also great. I personally see features before pep8 compliance. Especially as this isn't an important application.(no one depends on it) If you want to further help with this project, and do pep compliance, i can also start to do more of it.

I do like the idea of a requirement.txt . But I think something went wrong when you generated it and there are too many packages. I assume you used the pip command to autogenerate it. And I assume that took all packages that you have installed instead of the ones used in in this file. Or does mathplotlib have so many dependencies?

Thank you for your work.

**Comment by flameSla** [comment]
>I do like the idea of a requirement.txt . But I think something went wrong when you generated it and there are too many packages. I assume you used the pip command to autogenerate it. And I assume that took all packages that you have installed instead of the ones used in in this file. Or does mathplotlib have so many dependencies?

**pipreqs**

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+certifi==2022.12.7
```
Comment (suisseWalter) [inline requirements.txt]: to many requirements

**Review by suisseWalter — APPROVED** [review]
lgtm

**Pull request merged**